pkg/report/linux_test.go

```diff
@@ -230,3 +235,187 @@ func TestLinuxSymbolizeLine(t *testing.T) {
+			},
+		},
+		{
+			input: "c031 <f5e8> f7bf fff7 00ff",
+			output: parsedOpcodes{
+				rawBytes:       []byte{0x31, 0xc0, 0xe8, 0xf5, 0xbf, 0xf7, 0xf7, 0xff, 0xff, 0x00},
+				decompileFlags: []DecompilerFlag{},
+				offset:         2,
+			},
+		},
+		{
+			input: "(33221100) 77665544",
+			output: parsedOpcodes{
+				rawBytes:       []byte{0x00, 0x11, 0x22, 0x33, 0x44, 0x55, 0x66, 0x77},
+				decompileFlags: []DecompilerFlag{},
+				offset:         0,
+			},
+		},
+	}
+
+	leReporter := prepareLinuxReporter(targets.AMD64, t)
+	runValidTests(leReporter, validLeTests)
+	runInvalidTests(leReporter, invalidTests)
+
+	// Big endian tests.
+	validBeTests := []validTest{
+		{
+			input: "31 c0 <e8> f5 bf f7 ff",
+			output: parsedOpcodes{
+				rawBytes:       []byte{0x31, 0xc0, 0xe8, 0xf5, 0xbf, 0xf7, 0xff},
+				decompileFlags: []DecompilerFlag{},
+				offset:         2,
+			},
+		},
+		{
+			input: "31c0 <e8f5> bff5 f7ff ff00",
+			output: parsedOpcodes{
+				rawBytes:       []byte{0x31, 0xc0, 0xe8, 0xf5, 0xbf, 0xf5, 0xf7, 0xff, 0xff, 0x00},
+				decompileFlags: []DecompilerFlag{},
+				offset:         2,
+			},
+		},
+		{
+			input: "<00112233> 44556677",
+			output: parsedOpcodes{
+				rawBytes:       []byte{0x00, 0x11, 0x22, 0x33, 0x44, 0x55, 0x66, 0x77},
+				decompileFlags: []DecompilerFlag{},
+				offset:         0,
+			},
+		},
+	}
+
+	beReporter := prepareLinuxReporter(targets.S390x, t)
+	runValidTests(beReporter, validBeTests)
+	runInvalidTests(leReporter, invalidTests)
+
+	// Test ARM Thumb detection.
+
+	validArmTests := []validTest{
+		{
+			input: "0011 (2233) 4455",
+			output: parsedOpcodes{
+				rawBytes:       []byte{0x11, 0x00, 0x33, 0x22, 0x55, 0x44},
+				decompileFlags: []DecompilerFlag{FlagForceArmThumbMode},
+				offset:         2,
+			},
+		},
+		{
+			input: "(33221100) 77665544",
+			output: parsedOpcodes{
+				rawBytes:       []byte{0x00, 0x11, 0x22, 0x33, 0x44, 0x55, 0x66, 0x77},
+				decompileFlags: []DecompilerFlag{},
+				offset:         0,
+			},
+		},
+	}
+
+	armReporter := prepareLinuxReporter(targets.ARM, t)
+	runValidTests(armReporter, validArmTests)
+}
+
+func TestDisassemblyInReports(t *testing.T) {
+	archPath := filepath.Join("testdata", "linux", "decompile")
+	subFolders, err := ioutil.ReadDir(archPath)
+	if err != nil {
+		t.Fatalf("disassembly reports failed: %v", err)
+	}
+
+	runTest := func(reporter *linux, testFilePrefix string) func(t *testing.T) {
+		return func(t *testing.T) {
+			t.Parallel()
+
+			input, err := ioutil.ReadFile(testFilePrefix + ".in")
+			if err != nil {
+				t.Fatalf("failed to read input file: %v", err)
+			}
+			output, err := ioutil.ReadFile(testFilePrefix + ".out")
+			if err != nil {
+				t.Fatalf("failed to read output file: %v", err)
+			}
+
+			result := reporter.decompileReportOpcodes(input)
+			ioutil.WriteFile(testFilePrefix+".out", result, 0640)
+
+			if !bytes.Equal(output, result) {
+				t.Fatalf("Expected:\n%v\nGot:\n%v\n", string(output), string(result))
```

Comment:
You can avoid converting variables to string if you use `%s`, it automatically prints as string.